### ISSUE TITLE
Save button is invisible with Goldfish colorset chosen

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#4415]
**Steps To Reproduce**

To reproduce the behavior:
1. Go to `/admin/appearance/settings/basis` and set Goldfish color scheme.
2. Go to any page with submission button to find out the _Save_ button is invisible:

![empty-button](https://user-images.githubusercontent.com/56604405/82710884-3721f100-9c41-11ea-94f8-23ef277c3689.jpg)

![invisible](https://user-images.githubusercontent.com/56604405/82710887-39844b00-9c41-11ea-89bf-a2e56b2017fb.jpg)

**Comment by Ammko128** [comment]
@stpaultim 
Hi, I noticed that color schemes use the 'base' color from 'core/themes/basis/color/color.inc' as the background for the primary button. In goldfish this 'base' color is set to white (#fffeff) and since the background is set to #ffffff in goldfish, this primary button is invisible. One solution is to change the 'base' color of goldfish which I did in my PR. The other approach that I have in my mind is to add another color to all themes that will decide the background color of all primary buttons. I can change the code so primary buttons get another color if you consider that as a better approach than changing the base color of goldfish scheme.

Those were my suggestions for fixing this issue. I hope they are helpful. Have a good day. 👍⚪

**Comment by klonos** [comment]
Thanks for working on this issue @Ammko128 🙏 ...as it turns out, the same issue is present with the "Ice" preset as well.

Your PR changes a color used by the Basis theme, which is used as a base theme by many other themes. This makes me worry that we might break existing sites. I would start by looking into why the other presets don't have the same issue. If we figure that out, we can mimic whatever is done there for the Goldfish and Ice presets.

**Comment by Ammko128** [comment]
@klonos 
Hi, me again. I tried using Ice colorset and I noticed it doesn't have the exact same problem, but rather the save button disappears when you hover your mouse over the button.
As I noticed, all the colorsets use their 'base' color as color of their primary buttons, and 'hovermenu' color as color of that same button when it's hovered over. Also all the colorsets use their 'bg' color as a background color. By inspecting the array of Ice colorset, you can notice that it has the exact same color set as 'hovermenu' and 'bg' so that's why the button disappears when hovered over. It actually doesn't disappear but rather blends in with the color of the background.
Also when you do the same with Goldfish colorset, you should see that 'base' and 'bg' are almost same (1 hex value off). So that's why you can't see that save button until you hover the mouse over it and it changes color.
By looking at all the other colorsets, you can see that neither 'base' nor 'hovermenu' is same as it's 'bg' color. That's why all the other colorsets have their save button rendered correctly at all times.

![ice](https://user-images.githubusercontent.com/60924455/94401615-28a59200-016b-11eb-97d3-4a4c26603663.png)

![goldfish](https://user-images.githubusercontent.com/60924455/94401618-293e2880-016b-11eb-8c0e-43af841286eb.png)

I hope this was helpful in future decision making about this issue, since I don't think it can be fixed without breaking some sites that already use goldfish and ice presets.
And my PR didn't actually change the whole Basis theme. it only changed it's goldfish colorset. 👍

**Comment by olafgrabienski** [comment]
@Ammko128 Thanks for your detailed feedback!

That the PR may "break existing sites" isn't necessarily bad. It's just not clear how we deal with such situations, see https://github.com/backdrop/backdrop-issues/issues/4167.

**Comment by stpaultim** [comment]
To see this issue, you need to set the admin theme to "Basis". I updated the original issue to reflect this.

**Comment by stpaultim** [comment]
As I look at this issue closer, I think that the problem is a bit bigger. The color module goes through entire CSS files and replaces colors when you change color schemes. I believe that this is resulting in color changes that were not necessarily intended by the color module. 

I suspect that this is mostly an issue if you are using Basis as your admin theme or if users without access to the admin theme are accessing admin pages (node creation forms). 

I'm thinking about opening an issue where we prevent this from happening. @jenlampton and I just discussed two possibilities.

**Comment by indigoxela** [comment]
The problem is, that the color for the "mobile site background" is used in many other places. A quick fix could be to use `#d46f2e` (a dark orange) for that, so the buttons are visible. This might need some more testing, though...

![goldfish-visible-buttons](https://user-images.githubusercontent.com/842066/126270777-4441aaea-02bb-4285-a930-ee8b3f5a8100.png)

**Comment by klonos** [comment]
Wondering if a way forward with this could be #4829 and then converting the color presets in `core/themes/basis/color/color.inc` into proper settings/config 🤔  ...we could then ship core with a fixed version, but not touch config for existing sites.

**Comment by roxy-jaqkar** [comment]
This issue is also present in Views Slideshow navigation.

**Comment by herbdool** [comment]
I agree with @indigoxela on keeping the fix small and specific to just this for now.
